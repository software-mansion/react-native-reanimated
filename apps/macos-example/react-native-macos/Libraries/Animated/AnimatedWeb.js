/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow strict-local
 */

'use strict';

import AnimatedImplementation from './AnimatedImplementation';

export default {
  ...AnimatedImplementation,
  /* $FlowFixMe[incompatible-call] createAnimatedComponent expects to receive
   * types. Plain intrinsic components can't be typed like this */
  div: (AnimatedImplementation.createAnimatedComponent('div'): $FlowFixMe),
  /* $FlowFixMe[incompatible-call] createAnimatedComponent expects to receive
   * types. Plain intrinsic components can't be typed like this */
  span: (AnimatedImplementation.createAnimatedComponent('span'): $FlowFixMe),
  /* $FlowFixMe[incompatible-call] createAnimatedComponent expects to receive
   * types. Plain intrinsic components can't be typed like this */
  img: (AnimatedImplementation.createAnimatedComponent('img'): $FlowFixMe),
};
