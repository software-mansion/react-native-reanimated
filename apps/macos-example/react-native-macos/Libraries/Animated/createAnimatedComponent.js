/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 */

import View from '../Components/View/View';
import useMergeRefs from '../Utilities/useMergeRefs';
import useAnimatedProps from './useAnimatedProps';
import * as React from 'react';

export type AnimatedComponentType<
  -Props: {+[string]: mixed, ...},
  +Instance = mixed,
> = React.AbstractComponent<
  $ObjMap<
    Props &
      $ReadOnly<{
        passthroughAnimatedPropExplicitValues?: React.ElementConfig<
          typeof View,
        >,
      }>,
    () => any,
  >,
  Instance,
>;

export default function createAnimatedComponent<TProps: {...}, TInstance>(
  Component: React.AbstractComponent<TProps, TInstance>,
): AnimatedComponentType<TProps, TInstance> {
  return React.forwardRef((props, forwardedRef) => {
    const [reducedProps, callbackRef] = useAnimatedProps<TProps, TInstance>(
      // $FlowFixMe[incompatible-call]
      props,
    );
    // $FlowFixMe[incompatible-call]
    const ref = useMergeRefs<TInstance | null>(callbackRef, forwardedRef);

    // Some components require explicit passthrough values for animation
    // to work properly. For example, if an animated component is
    // transformed and Pressable, onPress will not work after transform
    // without these passthrough values.
    // $FlowFixMe[prop-missing]
    const {passthroughAnimatedPropExplicitValues, style} = reducedProps;
    const {style: passthroughStyle, ...passthroughProps} =
      passthroughAnimatedPropExplicitValues ?? {};
    const mergedStyle = {...style, ...passthroughStyle};

    return (
      <Component
        {...reducedProps}
        {...passthroughProps}
        style={mergedStyle}
        ref={ref}
      />
    );
  });
}
