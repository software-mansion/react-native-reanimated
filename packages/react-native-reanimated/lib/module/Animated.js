'use strict';

export { ReanimatedFlatList as FlatList } from './component/FlatList';
export { AnimatedImage as Image } from './component/Image';
export { AnimatedScrollView as ScrollView } from './component/ScrollView';
export { AnimatedText as Text } from './component/Text';
export { AnimatedView as View } from './component/View';
export { addWhitelistedNativeProps, addWhitelistedUIProps } from './ConfigHelper';
export { createAnimatedComponent } from './createAnimatedComponent';
//# sourceMappingURL=Animated.js.map