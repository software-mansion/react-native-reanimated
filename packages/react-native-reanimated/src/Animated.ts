'use strict';
import type { Extrapolate as _Extrapolate } from './interpolateColor';
import type {
  SharedValue as _SharedValue,
  AnimatedTransform as _AnimatedTransform,
  AnimateStyle as _AnimateStyle,
  StylesOrDefault as _StylesOrDefault,
  EasingFunction as _EasingFunction,
} from './commonTypes';
import type { DerivedValue as _DerivedValue } from './hook/useDerivedValue';
import type {
  TransformStyleTypes as _TransformStyleTypes,
  Adaptable as _Adaptable,
  AdaptTransforms as _AdaptTransforms,
  AnimateProps as _AnimateProps,
} from './helperTypes';

import type { AnimatedScrollViewProps as _AnimatedScrollViewProps } from './component/ScrollView';
import type { FlatListPropsWithLayout as _FlatListPropsWithLayout } from './component/FlatList';

export { createAnimatedComponent } from './createAnimatedComponent';
export { AnimatedText as Text } from './component/Text';
export { AnimatedView as View } from './component/View';
export { AnimatedScrollView as ScrollView } from './component/ScrollView';
export { AnimatedImage as Image } from './component/Image';
export { ReanimatedFlatList as FlatList } from './component/FlatList';
export {
  addWhitelistedNativeProps,
  addWhitelistedUIProps,
} from './ConfigHelper';
/**
 * @deprecated Please import `Extrapolate` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type Extrapolate = typeof _Extrapolate;
/**
 * @deprecated Please import `SharedValue` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */

export type SharedValue<T> = _SharedValue<T>;
/**
 * @deprecated Please import `DerivedValue` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type DerivedValue<T> = _DerivedValue<T>;
/**
 * @deprecated Please import `Adaptable` directly from `react-native-reanimated`
 *   instead of `Animated` namespace.
 */
export type Adaptable<T> = _Adaptable<T>;
/**
 * @deprecated Please import `TransformStyleTypes` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type TransformStyleTypes = _TransformStyleTypes;
/**
 * @deprecated Please import `AdaptTransforms` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AdaptTransforms<T> = _AdaptTransforms<T>;
/**
 * @deprecated Please import `AnimatedTransform` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimatedTransform = _AnimatedTransform;
/**
 * @deprecated Please import `AnimateStyle` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimateStyle<S> = _AnimateStyle<S>;
/**
 * @deprecated Please import `StylesOrDefault` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type StylesOrDefault<S> = _StylesOrDefault<S>;
/**
 * @deprecated Please import `AnimateProps` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimateProps<P extends object> = _AnimateProps<P>;
/**
 * @deprecated Please import `EasingFunction` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type EasingFunction = _EasingFunction;
/**
 * @deprecated Please import `AnimatedScrollViewProps` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimatedScrollViewProps = _AnimatedScrollViewProps;
/**
 * @deprecated Please import `FlatListPropsWithLayout` directly from
 *   `react-native-reanimated` instead of `Animated` namespace.
 */
export type FlatListPropsWithLayout<T> = _FlatListPropsWithLayout<T>;
