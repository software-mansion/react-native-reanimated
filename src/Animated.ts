import type { Extrapolate as _Extrapolate } from './reanimated2/interpolateColor';
import type { SharedValue as _SharedValue } from './reanimated2/commonTypes';
import type { DerivedValue as _DerivedValue } from './reanimated2/hook/useDerivedValue';
import type {
  TransformStyleTypes as _TransformStyleTypes,
  Adaptable as _Adaptable,
  AdaptTransforms as _AdaptTransforms,
  AnimatedTransform as _AnimatedTransform,
  AnimateStyle as _AnimateStyle,
  StylesOrDefault as _StylesOrDefault,
  AnimateProps as _AnimateProps,
} from './reanimated2/helperTypes';
import type { EasingFunction as _EasingFunction } from './reanimated2/Easing';
import {
  addWhitelistedNativeProps as _addWhitelistedNativeProps,
  addWhitelistedUIProps as _addWhitelistedUIProps,
} from './ConfigHelper';
import type { AnimatedScrollViewProps as _AnimatedScrollViewProps } from './reanimated2/component/ScrollView';
import type { FlatListPropsWithLayout as _FlatListPropsWithLayout } from './reanimated2/component/FlatList';
export { default as createAnimatedComponent } from './createAnimatedComponent';

export { AnimatedText as Text } from './reanimated2/component/Text';
export { AnimatedView as View } from './reanimated2/component/View';
export { AnimatedScrollView as ScrollView } from './reanimated2/component/ScrollView';
export { AnimatedImage as Image } from './reanimated2/component/Image';
export { ReanimatedFlatList as FlatList } from './reanimated2/component/FlatList';
/**
 * @deprecated Please import `Extrapolate` directly from `react-native-reanimated` instead of `Animated` namespace.
 */
export type Extrapolate = typeof _Extrapolate;
/**
 * @deprecated Please import `SharedValue` directly from `react-native-reanimated` instead of `Animated` namespace.
 */

export type SharedValue<T> = _SharedValue<T>;
/**
 * @deprecated Please import `DerivedValue` directly from `react-native-reanimated` instead of `Animated` namespace.
 */
export type DerivedValue<T> = _DerivedValue<T>;
/**
 * @deprecated Please import `Adaptable` directly from `react-native-reanimated` instead of `Animated` namespace.
 */
export type Adaptable<T> = _Adaptable<T>;
/**
 * @deprecated Please import `TransformStyleTypes` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type TransformStyleTypes = _TransformStyleTypes;
/**
 * @deprecated Please import `AdaptTransforms` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type AdaptTransforms<T> = _AdaptTransforms<T>;
/**
 * @deprecated Please import `AnimatedTransform` directly from `react-native-reanimated` instead of `Animated` namespace.
 */
export type AnimatedTransform = _AnimatedTransform;
/**
 * @deprecated Please import `AnimateStyle` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type AnimateStyle<S> = _AnimateStyle<S>;
/**
 * @deprecated Please import `StylesOrDefault` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type StylesOrDefault<S> = _StylesOrDefault<S>;
/**
 * @deprecated Please import `AnimateProps` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type AnimateProps<P extends object> = _AnimateProps<P>;
/**
 * @deprecated Please import `EasingFunction` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type EasingFunction = _EasingFunction;
/**
 * @deprecated Please import `addWhitelistedNativeProps` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export const addWhitelistedNativeProps = _addWhitelistedNativeProps;
/**
 * @deprecated Please import `addWhitelistedUIProps` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export const addWhitelistedUIProps = _addWhitelistedUIProps;
/**
 * @deprecated Please import `AnimatedScrollViewProps` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type AnimatedScrollViewProps = _AnimatedScrollViewProps;
/**
 * @deprecated Please import `FlatListPropsWithLayout` directly from `react-native-reanimated` instead of `Animated` namespace.
 * */
export type FlatListPropsWithLayout<T> = _FlatListPropsWithLayout<T>;
