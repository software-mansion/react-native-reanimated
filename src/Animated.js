import { LogBox } from 'react-native';
import createAnimatedComponent from './createAnimatedComponent';
import {
  addWhitelistedNativeProps,
  addWhitelistedUIProps,
} from './ConfigHelper';
import * as reanimated1 from './reanimated1';
import ReanimatedComponents from './reanimated2/component';

const Animated = {
  // components
  ...ReanimatedComponents,
  createAnimatedComponent,
  // configuration
  addWhitelistedNativeProps,
  addWhitelistedUIProps,
  // reanimated 1
  ...reanimated1,
};

export * from './reanimated2';
export * from './reanimated1';
export default Animated;

// I think we can ignore this message as long as Gesture Handler doesn't
// try to load the Reanimated module. I figured it should be here instead of
// RNGH because this prevents the message from being displayed for all
// versions of RNGH.
LogBox.ignoreLogs([
  'RCTBridge required dispatch_sync to load RNGestureHandlerModule. This may lead to deadlocks',
]);
