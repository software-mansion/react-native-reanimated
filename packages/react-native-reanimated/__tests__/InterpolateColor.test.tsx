import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Button, View } from 'react-native';

import Animated, {
  interpolateColor,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from '../src';

describe('colors interpolation', () => {
  test('interpolates rgb without gamma correction', () => {
    const colors = ['#105060', '#609020'];
    let interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'RGB', {
      gamma: 1,
    });
    expect(interpolatedColor).toBe('rgba(56, 112, 64, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'RGB', {
      gamma: 1,
    });
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'RGB', {
      gamma: 1,
    });
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 1)');
  });

  test('interpolates rgb with gamma correction', () => {
    const colors = ['#105060', '#609020'];

    let interpolatedColor = interpolateColor(0.5, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(71, 117, 73, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 1)');
  });

  test('interpolates hsv', () => {
    let colors = ['#105060', '#609020'];
    let interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(23, 120, 54, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 1)');

    colors = ['#ff4800', '#ff00fb'];
    interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(0, 255, 166, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(255, 72, 0, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'HSV', {
      useCorrectedHSVInterpolation: false,
    });
    expect(interpolatedColor).toBe('rgba(255, 0, 251, 1)');
  });

  test('interpolates corrected hsv', () => {
    let colors = ['#105060', '#609020'];
    let interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(23, 120, 54, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 1)');

    colors = ['#ff4800', '#ff00fb'];
    interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(255, 0, 90, 1)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(255, 72, 0, 1)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(255, 0, 251, 1)');
  });

  test('interpolates css colors', () => {
    const colors = ['red', 'green'];
    const interpolatedColor = interpolateColor(0.5, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(186, 93, 0, 1)');
  });

  test('interpolates semi-transparent colors', () => {
    const colors = ['#10506050', '#60902070'];
    let interpolatedColor = interpolateColor(0.5, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(71, 117, 73, 0.376)');

    interpolatedColor = interpolateColor(0, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 0.314)');

    interpolatedColor = interpolateColor(1, [0, 1], colors);
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 0.439)');

    interpolatedColor = interpolateColor(0.5, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(23, 120, 54, 0.376)');

    interpolatedColor = interpolateColor(0, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(16, 80, 96, 0.314)');

    interpolatedColor = interpolateColor(1, [0, 1], colors, 'HSV');
    expect(interpolatedColor).toBe('rgba(96, 144, 32, 0.439)');
  });

  test('handles tiny values', () => {
    const colors = ['#00000000', '#ff802001'];

    // We don't want output like "rgba(4, 2, 0, 3.921568627450981e-7)":
    const interpolatedColor = interpolateColor(0.0001, [0, 1], colors);
    expect(interpolatedColor).toBe(`rgba(4, 2, 0, 0)`);
  });

  describe('simple transparent to color interpolation', () => {
    const cases = [
      {
        name: 'transparent to color at midpoint',
        value: 0.5,
        inputRange: [0, 1],
        outputRange: ['transparent', '#ff0000'],
        expected: 'rgba(255, 0, 0, 0.5)',
      },
      {
        name: 'transparent at start position',
        value: 0,
        inputRange: [0, 1],
        outputRange: ['transparent', '#ff0000'],
        expected: 'rgba(255, 0, 0, 0)',
      },
      {
        name: 'color at end position',
        value: 1,
        inputRange: [0, 1],
        outputRange: ['transparent', '#ff0000'],
        expected: 'rgba(255, 0, 0, 1)',
      },
      {
        name: 'transparent to transparent',
        value: 0.5,
        inputRange: [0, 1],
        outputRange: ['transparent', 'transparent'],
        expected: 'rgba(0, 0, 0, 0)',
      },
    ];

    const colorSpaces: Array<{
      colorSpace: 'RGB' | 'HSV' | 'LAB';
      options?: Record<string, unknown>;
      eps?: number;
    }> = [
      { colorSpace: 'RGB' },
      { colorSpace: 'RGB', options: { gamma: 1 } },
      { colorSpace: 'HSV' },
      { colorSpace: 'HSV', options: { useCorrectedHSVInterpolation: false } },
      // LAB may produce slightly different results, but the differences are usually small
      { colorSpace: 'LAB', eps: 1e-5 },
    ];

    colorSpaces.forEach(({ colorSpace, options, eps }) => {
      test.each(cases)(
        `$name using ${colorSpace}${options ? ` with options ${JSON.stringify(options)}` : ''}`,
        ({ value, inputRange, outputRange, expected }) => {
          const result = interpolateColor(
            value,
            inputRange,
            outputRange,
            colorSpace,
            options
          );

          if (eps) {
            const getChannels = (color: string) =>
              color
                .replace('rgba(', '')
                .replace(')', '')
                .split(',')
                .map((v) => parseFloat(v.trim()));

            getChannels(result).forEach((v, i) => {
              expect(v).toBeCloseTo(getChannels(expected)[i], eps);
            });
          } else {
            expect(result).toBe(expected);
          }
        }
      );
    });
  });

  describe('color interpolation with multiple transparent colors', () => {
    const inputRange = [0, 0.2, 0.4, 0.6, 0.8, 1];
    const outputRange = [
      'transparent',
      'transparent',
      'red',
      'transparent',
      'blue',
      '#00ff00',
    ];

    const cases: [number, string][] = [
      [0.1, 'rgba(255, 0, 0, 0)'], // red transparent
      [0.2, 'rgba(255, 0, 0, 0)'], // red transparent
      [0.3, 'rgba(255, 0, 0, 0.5)'], // between transparent red and red
      [0.4, 'rgba(255, 0, 0, 1)'], // red
      [0.5, 'rgba(255, 0, 0, 0.5)'], // between red and transparent red
      [0.6, 'rgba(255, 0, 0, 0)'], // red transparent
      [0.6000001, 'rgba(0, 0, 255, 0)'], // blue transparent
      [0.7, 'rgba(0, 0, 255, 0.5)'], // between transparent blue and blue
      [0.8, 'rgba(0, 0, 255, 1)'], // blue
      [0.9, 'rgba(0, 127.5, 127.5, 1)'], // between blue and green
      [1, 'rgba(0, 255, 0, 1)'], // green
    ];

    test.each(cases)(`for value %s, the result is %s`, (value, expected) => {
      expect(
        interpolateColor(value, inputRange, outputRange, 'RGB', {
          gamma: 1,
        })
      ).toBe(expected);
    });
  });

  function TestComponent() {
    const color = useSharedValue('#105060');

    const animatedStyle = useAnimatedStyle(() => {
      return {
        backgroundColor: withTiming(color.value, { duration: 500 }),
      };
    });

    return (
      <View>
        <Animated.View style={animatedStyle} testID="view" />
        <Button
          onPress={() => {
            color.value = '#609020';
          }}
          title="run animation"
          testID="button"
        />
      </View>
    );
  }

  test('interpolates with withTiming animation', () => {
    jest.useFakeTimers();

    const { getByTestId } = render(<TestComponent />);
    const view = getByTestId('view');
    const button = getByTestId('button');

    expect(view).toHaveAnimatedStyle(
      { backgroundColor: '#105060' },
      { shouldMatchAllProps: true }
    );

    fireEvent.press(button);
    jest.advanceTimersByTime(250);

    expect(view).toHaveAnimatedStyle(
      { backgroundColor: 'rgba(71, 117, 73, 1)' },
      { shouldMatchAllProps: true }
    );

    jest.advanceTimersByTime(250);

    expect(view).toHaveAnimatedStyle(
      { backgroundColor: 'rgba(96, 144, 32, 1)' },
      { shouldMatchAllProps: true }
    );

    jest.runOnlyPendingTimers();
    jest.useRealTimers();

    const rendered = render(<TestComponent />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
