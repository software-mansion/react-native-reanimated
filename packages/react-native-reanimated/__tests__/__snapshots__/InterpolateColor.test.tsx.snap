// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors interpolation interpolates with withTiming animation 1`] = `
<View>
  <View
    collapsable={false}
    jestAnimatedProps={
      {
        "value": {},
      }
    }
    jestAnimatedStyle={
      {
        "value": {
          "backgroundColor": "#105060",
        },
      }
    }
    jestInlineStyle={{}}
    style={
      [
        {
          "backgroundColor": "#105060",
        },
      ]
    }
    testID="view"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        run animation
      </Text>
    </View>
  </View>
</View>
`;
